src/effects/builtin/common/abstracteffectmodel.cpp

```diff
@@ -42,42 +44,35 @@ bool AbstractEffectModel::inited() const
 
 std::shared_ptr<au::effects::EffectInstance> AbstractEffectModel::instance() const
 {
-    EffectInstanceId id = this->instanceId();
-    if (id == 0) {
-        return nullptr;
-    }
-
-    return instancesRegister()->instanceById(id);
+    return instancesRegister()->instanceById(instanceId());
 }
 
-const EffectSettings* AbstractEffectModel::settings() const
+const EffectSettings& AbstractEffectModel::settings() const
 {
-    EffectInstanceId id = this->instanceId();
-    if (id == 0) {
-        return nullptr;
+    const EffectSettings* s = instancesRegister()->settingsById(instanceId());
+    IF_ASSERT_FAILED(s) {
+        static EffectSettings null;
+        return null;
     }
 
-    return instancesRegister()->settingsById(id);
+    return *s;
 }
 
 EffectSettingsAccessPtr AbstractEffectModel::settingsAccess() const
 {
-    EffectInstanceId id = this->instanceId();
-    if (id == 0) {
-        return nullptr;
-    }
-
-    return instancesRegister()->settingsAccessById(id);
+    return instancesRegister()->settingsAccessById(instanceId());
 }
 
 EffectInstanceId AbstractEffectModel::instanceId() const
 {
-    return m_instanceId.toULongLong();
+    const auto id = m_inited ? m_instanceId : BuiltinEffectViewLoader::initializationInstanceId();
```

Comment:
If we have a guarantee that instance will be initialized at the object creation, why do we try to reinitialize it here?